api-server/server/boot/authentication.js

```diff
@@ -37,8 +37,14 @@ module.exports = function enableAuthentication(app) {
   app.enableAuth();
   const ifUserRedirect = ifUserRedirectTo();
   const ifNoUserRedirectHome = ifNoUserRedirectTo(homeLocation);
-  const saveAuthCookies = saveResponseAuthCookies();
-  const loginSuccessRedirect = loginRedirect();
+  const devSaveAuthCookies = devSaveResponseAuthCookies();
+  const devLoginSuccessRedirect = devLoginRedirect();
+  const addRedirect = (req, res, next) => {
+    if (req && req.query && req.query.returnTo) {
+      req.query.returnTo = `${homeLocation}/${req.query.returnTo}`;
+    }
+    return next();
+  };
```

Comment:
This part of the logic affects what gets executed here: https://github.com/freeCodeCamp/freeCodeCamp/pull/40161/files#diff-1f182080313f7e6aae86f53b770161759a3008abae3e3bdba965fadfd7ed4068R118 - unless I am totally missing on something.

Reply:
Just to be transparent (we discussed this elsewhere):

> Unlike a session, a query string can be added by anyone including client?

I can't see what the diff is referring to, but this is just a refactor.  The danger arises if we allow arbitrary directs, but here we control `homeLocation`.  But, yes, the `returnTo` string could be anything, so any change that allows redirection to other domains other than www.freecodecamp.org will need to be implemented carefully.

Reply:
Yes - that is correct. I fixed my comment above to point to the code where it can get picked up.